### ISSUE TITLE
Which prompt format works best for GOT-OCR for OCR

### ISSUE THREAD

**Issue by futureisatyourhand** [Ucas-HaoranWei/GOT-OCR2.0#166]
Thank your project， and I‘m interested in combining LLM and visual-encoder for OCR.  
For GOT-OCR2.0，I have a question, e.g., the impact of prompt，which prompt format works best roi which is cropped and contains texts ?
 
I have two different prompt, the result exists significant differences for only same roi image containing 4-5 numbers. These are

 conv_mpt = Conversation(
     system="""<|im_start|>system
 - You are a helpful language and vision assistant.
 - You are able to understand the visual content that the user provides, and assist the user with a variety of tasks using natural language.
- You should follow the instructions carefully and explain your answers in detail.""",
   roles=("<|im_start|>user\n", "<|im_start|>assistant\n"),
     version="mpt",
     messages=(),
     offset=0,
     sep_style=SeparatorStyle.MPT,
     sep="<|im_end|>",
 )

and

conv_mpt = Conversation(
    system="""<|im_start|>system
You should follow the instructions carefully and explain your answers in detail.""",
    # system = None,
    roles=("<|im_start|>user\n", "<|im_start|>assistant\n"),
    version="mpt",
    messages=(),
    offset=0,
    sep_style=SeparatorStyle.MPT,
    sep="<|im_end|>",
)
 
Because of the different results, do you have any suggestions for using prompts in GOT-OCR 2.0?

Look forward to your reply！

**Comment by Ucas-HaoranWei** [comment]
Each one is OK, it's best to keep it consistent with  GOT.

**Comment by futureisatyourhand** [comment]
> Each one is OK, it's best to keep it consistent with GOT.

Thanks，I fine-tune model on our dataset. The accuracy is perfect for OCR.

**Comment by thhung** [comment]
@futureisatyourhand Could you share how do you manage to finetune on your dataset? I currently has some problem with my training. So I am trying to gather info as much as possible. Thanks in advance.

**Comment by futureisatyourhand** [comment]
> @futureisatyourhand Could you share how do you manage to finetune on your dataset? I currently has some problems with my training. So I am trying to gather info as much as possible. Thanks in advance.

I use command from Github. Firstly, you should generate train datasets with Prompt and targets. Then, you can update code according to your task, such as the way to preprocess image datasets. In fact, for OCR, I write py file to generate training data. I use prompt is:

   dicts["image"]=filepath
    dicts["conversations"]=[{"from":"human","value":"<image>\nOCR:"},
                                        ("from":"gpt","value":"<ref>A</ref>")]
  train_data.append(dicts)
note:A represents content of current image.
Final, I save train_data to file with JSON format.

I train 1epoch, 8batch size, gradient_accumulation_steps 8. My datasets is 600w.

**Comment by thhung** [comment]
> 600w

Many thanks. Just to confirm: 600w = 6 000 000 data points, right? That is a really big dataset for my scale.

**Comment by futureisatyourhand** [comment]
> > 600w
> 
> Many thanks. Just to confirm: 600w = 6 000 000 data points, right? That is a really big dataset for my scale.

Yes, 6M images with content.
